server/src/QRServer/db/connector.py

```diff
@@ -288,13 +303,70 @@ async def get_ranking(self, start_date: datetime, end_date: datetime, ranked_onl
         await c.execute(
             "select"
             " u.username,"
+            " r.user_id,"
+            " wins,"
+            " total_games"
+            " from rankings r"
+            " inner join users u on u.id = r.user_id"
+            " where r.year == ?"
+            " and r.month == ?"
+            " order by r.position asc"  # Lower position - higher in leaderboards
+            " limit 100", (
+                start_date.year,
+                start_date.month,
+            )
+        )
+
+        ranking_entries = []
+        rows = await c.fetchall()
+        for row in rows:
+            ranking_entries.append(RankingEntry(
+                username=row[0],
+                user_id=row[1],
+                wins=row[2],
+                games=row[3],
+            ))
+        return ranking_entries
+
+    async def update_ranking(self, rankings: List[RankingEntry], year: int, month: int) -> None:
+        for position, entry in enumerate(rankings):
+            c = await self.conn.cursor()
+            await c.execute(
+                "insert or replace into rankings ("
+                "  year,"
+                "  month,"
+                "  position,"
+                "  user_id,"
+                "  wins,"
+                "  total_games"
+                ") values ("
+                "?, ?, ?, ?, ?, ?"
+                ")", (
+                    year,
+                    month,
+                    position+1,
+                    entry.user_id,
+                    entry.wins,
+                    entry.games,
+                ))
+        await self.conn.commit()
+
+    async def _generate_ranking(self, start_date: datetime, end_date: datetime, ranked_only=True,
+                                include_void=False) -> List[RankingEntry]:
```

Comment:
Why accept those values as parameters when you can just get them here?

Reply:
what do you mean? as in generate datetime?

Reply:
Use config values for `ranked_only` and `include_void`